impl/httpMetricsManager.go

```diff
@@ -30,10 +68,36 @@ func (m *HttpMetricsManager) GetThroughput() int {
 	return m.requestSent
 }
 
+// return the sampletime(duration) at input percentile
+func (m *HttpMetricsManager) GetDurationAt(thre int) time.Duration {
+	if !sort.IsSorted(m.records) {
+		sort.Sort(m.records)
+	}
```

Comment:
In long run, yes. I'm thinking about other way to circumvent sorting all records. One way I thought of is to divide the records into bins(0~10%, 10%~20%, etc..) so that it trades of accuracy for speed. But typical metrics percentiles are 90%,95% and 99% which may cause binning more complicated and inefficient.